job/server/src/main/java/alluxio/job/replicate/MoveDefinition.java

```diff
@@ -0,0 +1,129 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * A job to move a block. This job is invoked by the checker of replication level in
+ * FileSystemMaster.
+ */
+@NotThreadSafe
+public final class MoveDefinition
+    extends AbstractVoidJobDefinition<MoveConfig, SerializableVoid> {
+  private static final Logger LOG = LoggerFactory.getLogger(MoveDefinition.class);
+
+  /**
+   * Constructs a new {@link MoveDefinition}.
+   */
+  public MoveDefinition() {
+  }
+
+  @Override
+  public Class<MoveConfig> getJobConfigClass() {
+    return MoveConfig.class;
+  }
+
+  @Override
+  public Map<WorkerInfo, SerializableVoid> selectExecutors(MoveConfig config,
+      List<WorkerInfo> jobWorkerInfoList, SelectExecutorsContext context) {
+    Preconditions.checkArgument(!jobWorkerInfoList.isEmpty(), "No worker is available");
+
+    String workerHost = config.getWorkerHost();
+
+    Map<WorkerInfo, SerializableVoid> result = Maps.newHashMap();
+
+    Collections.shuffle(jobWorkerInfoList);
+    for (WorkerInfo workerInfo : jobWorkerInfoList) {
+      // Select job workers that have this block locally to move
+      if (workerHost.equals(workerInfo.getAddress().getHost())) {
+        result.put(workerInfo, null);
+        return result;
+      }
+    }
+    return result;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * This task will evict the given block.
+   */
+  @Override
+  public SerializableVoid runTask(MoveConfig config, SerializableVoid args, RunTaskContext context)
+      throws Exception {
+    AlluxioBlockStore blockStore = AlluxioBlockStore.create(context.getFsContext());
+
+    long blockId = config.getBlockId();
+    String localHostName = NetworkAddressUtils.getConnectHost(ServiceType.WORKER_RPC,
+        ServerConfiguration.global());
+    List<BlockWorkerInfo> workerInfoList = blockStore.getAllWorkers();
+    WorkerNetAddress localNetAddress = null;
+
+    for (BlockWorkerInfo workerInfo : workerInfoList) {
+      if (workerInfo.getNetAddress().getHost().equals(localHostName)) {
+        localNetAddress = workerInfo.getNetAddress();
+        break;
+      }
+    }
+    if (localNetAddress == null) {
+      String message = String.format("Cannot find a local block worker to move block %d", blockId);
+      throw new NotFoundException(message);
+    }
+    BlockStoreLocationProto srcProto = config.getSource().toProto();
+    BlockStoreLocationProto dstProto = config.getDestination().toProto();
+
+    MoveBlockRequest request = MoveBlockRequest.newBuilder().setBlockId(blockId).setSrc(srcProto)
+        .setDst(dstProto).build();
+    BlockWorkerClient blockWorker = null;
+    try {
+      blockWorker = context.getFsContext().acquireBlockWorkerClient(localNetAddress);
+      blockWorker.moveBlock(request);
+    } finally {
```

Comment:
Can change this to use `try-with-resources`?

Reply:
@yuzhu ping

Reply:
`blockWorker.close()` here does not seem to do what i want though, at least not the same.  
`context.getFsContext().releaseBlockWorkerClient` seems more appropriate?